### ISSUE TITLE
Add test to guarantee data is not reingested when the journald input is restarted

### ISSUE THREAD

**Issue by pierrehilbert** [elastic/beats#37876]
## Issue
Meta issue: https://github.com/elastic/beats/issues/37086
To have journald GA, we should ensure that we are covering key scenarios with enough testing.

## Acceptance Criteria 
Tests are in place to confirm:
- [x] We don't have data duplication when the input is restarted
- [x] We have an at least once delivery guarantee 
- [ ] Performances are comparable with filestream reading syslog

**Comment by belimawr** [comment]
@pierrehilbert I believe it does not make sense to compare the performance of the Journald input with Filestream reading from syslog, at least not as part of this issue that is about not duplicating data on restart.

If we need a comparison in performance between Filestream reading syslog logs and Journald input reading the same data, we can create a separated issue for this.

What do you think?

**Comment by pierrehilbert** [comment]
Agree this doesn't need to be covered in this issue.
I still think we should run a performance comparison to measure the impact and be able to share the information to our customers if this is relevant (and therefore highlight rsyslog as an alternative to avoid this performance impact).
WDYT?

**Comment by belimawr** [comment]
> Agree this doesn't need to be covered in this issue. I still think we should run a performance comparison to measure the impact and be able to share the information to our customers if this is relevant (and therefore highlight rsyslog as an alternative to avoid this performance impact). WDYT?

Yes, we can run those benchmarks, it's gonna be an interesting exercise.
